src/main/java/Parser.java

```diff
@@ -0,0 +1,172 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import dukeexception.DukeMissingDescriptionException;
+import dukeexception.DukeUnknownInputException;
+
+/**
+ * Deals with making sense of the user command.
+ */
+public class Parser {
+
+    /**
+     * Gets the type of command aka what the user wants like add or delete tasks.
+     * @param command The user command.
+     * @return The type of the command.
+     */
+    public static String getCommandType(String command) {
+        String[] splitted = command.split(" ", 2);
+        return splitted[0];
+    }
+
+    /**
+     * Gets the task number from the user command to be marked done.
+     *
+     * @param command User command.
+     * @return The task number to be marked done.
+     * @throws DukeMissingDescriptionException When there is no task number given.
+     * @thows DukeUnknownInputException When what user gives is not a number.
+     */
+    public static int markDoneNum(String command) {
+        if (command.length() <= 5) {
+            throw new DukeMissingDescriptionException("Task number missing.");
+        }
+        String num = command.substring(5);
+        try {
+            return Integer.valueOf(num);
+        } catch (NumberFormatException e) {
+            throw new DukeUnknownInputException("Need task NUMBER.");
+        }
+    }
+
+    /**
+     * Gets the description for the To Do task.
+     *
+     * @param command User command.
+     * @return The description of the To Do task.
+     * @throws DukeMissingDescriptionException If there is no description given.
+     */
+    public static String todoDescription(String command) {
+        if (command.length() <= 5) {
+            throw new DukeMissingDescriptionException("Todo description missing.");
+        }
+        return command.substring(5);
+    }
+
+    /**
+     * Gives the description and deadline of the Deadline task.
+     *
+     * @param command User command.
+     * @return A String[] where 0 index contains the description and 1 index contains the deadline.
+     * @throws DukeMissingDescriptionException When no description of deadline is given by user.
+     * @throws DukeUnknownInputException When /by is missing in command.
+     */
+    public static String[] deadlineParams(String command) {
```

Comment:
Some methods (like this one) can be better named. It is recommended to use verbs such as `getDescriptionsAndDeadlines`.